server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/management/RabbitMQManagementApi.java

```diff
@@ -62,13 +65,14 @@
 
     }
 
-    Stream<MailQueueName> listCreatedMailQueueNames() {
-        return api.listQueues()
-            .stream()
-            .map(x -> x.name)
-            .map(MailQueueName::fromRabbitWorkQueueName)
-            .flatMap(OptionalUtils::toStream)
-            .distinct();
+    public Stream<MailQueueName> listCreatedMailQueueNames() {
+        return requestExecutor.execute(
```

Comment:
you don't really want to retry the whole expression but only the api call, right ?